dev/PushNotifications/PushNotificationBackgroundTask.cpp

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+
+#include "pch.h"
+
+#include "PushNotificationBackgroundTask.h"
+
+#include <winrt/base.h>
+#include <winrt/Windows.ApplicationModel.Core.h>
+
+#include "PushNotificationReceivedEventArgs.h"
+#include "externs.h"
+
+namespace winrt
+{
+    using namespace Windows::ApplicationModel::Background;
+    using namespace Microsoft::Windows::PushNotifications;
+    using namespace Windows::ApplicationModel::Core;
+}
+
+wil::unique_handle g_waitHandleForArgs = wil::unique_handle(CreateEvent(nullptr, FALSE, FALSE, nullptr));
+
+void PushNotificationBackgroundTask::Run(winrt::IBackgroundTaskInstance const& taskInstance)
+{
+    auto appProperties = winrt::CoreApplication::Properties();
+    winrt::PushNotificationReceivedEventArgs activatedEventArgs = winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationReceivedEventArgs>(taskInstance);
+    appProperties.Insert(ACTIVATED_EVENT_ARGS_KEY, activatedEventArgs);
+
+    SetEvent(g_waitHandleForArgs.get());
```

Comment:
I see. I thought of `winrt::throw_last_error` but it returns the last error on the calling thread, where the background task happens in another thread. Do you have a suggestion on handling this scenario?

Reply:
It depends on what you want to happen if the task cannot run.

Maybe the best thing to do is to ignore the problem. After all, there might be something else that prevents the task from running even before we get to the Run() method. Maybe the platform deals with the exception in the Run() method by crashing the app? If so, then this is fine because the entire process is gone, so it doesn't matter what happens to the code in the process that was waiting for the activated event args.

We should at least comment which approach we're taking and why.

Reply:
If there's an issue before calling Run(), I don't think we have a much of a choice. I would expect that Background Infrastructure folks look into this kind of issue.

In the case we hit an issue during Run(), I think we should let the task to throw as it happens now. IIRC while prototyping this feature, if issues occurred at this point then the app crashes, which should be fine.